tools/docker/tests/test_flows.sh

```diff
@@ -0,0 +1,106 @@
+#find prplmesh folder
+scriptdir="$(cd "${0%/*}"; pwd)"
+installdir="${scriptdir%/*/*/*/*}/build/install"
+
+dbg() {
+    [ "$VERBOSE" = "true" ] && echo "$@"
+}
+
+status() {
+	printf '\033[1;35m'"$@\n"'\033[0m'
+}
+
+err() {
+    printf '\033[1;31m'"$@\n"'\033[0m'
+}
+
+success() {
+    printf '\033[1;32m'"$@\n"'\033[0m'
+}
+
+run() {
+    dbg "$*"
+    "$@" || exit $?
+}
+
+report() {
+    msg="$1"; shift
+    if "$@"; then
+        success "OK $msg"
+    else
+        err "FAIL $msg"
+        error=1
+    fi
+}
+
+start_gw_repeater() {
+    dbg "delete running containers before starting test"
+    docker rm -f $(docker ps -a -q) >/dev/null 2>&1
+    dbg "start gw and repeater"
+    ${scriptdir}/test_gw_repeater.sh >/dev/null 2>&1
+}
+
+send_topology_query() {
+    dbg "Sending toplogy query to al_mac $1"
+    docker exec -it gateway ${installdir}/bin/beerocks_cli -c "bml_trigger_topology_discovery $1" >/dev/null 2>&1
+}
+
+check_log_for_query() {
+    dbg "Confirming topology query was received"
+    docker exec -it repeater cat tmp/$USER/beerocks/logs/beerocks_agent.log | grep -i -q "Topology Query"
+}
+
+test_topology() {
+    status "test topology query"
+    send_topology_query $1
+    check_log_for_query
+}
+
+test_init() {
+    status "test initialization"
+    start_gw_repeater || {
+        err "start GW+Repeater failed, abort"
+        exit 1
+    }
+    AL_MAC=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep IRE_BRIDGE | awk '{print $5}' | cut -d ',' -f 1)
```

Comment:
Ouch. After my changes for the non-Intel agent, there won't be a IRE_BRIDGE in the topology any more...

I was thinking that we should use fixed MAC addresses instead of the random ones we have now. We can pass a four-byte base MAC address as an argument to the runner, and extend it with predictable sub-addresses for the various interfaces we create within the container.

But that's something for a later PR.

Reply:
I did this and the change was merged, so you can be sure that the gateway's AL_MAC is 00:11:22:33:00:00 and the repeater's is aa:bb:cc:dd:00:00.

Note that you need to refresh your docker images for that (or pull them from dockerhub).

Reply:
Will fix in the next review iteration

Reply:
I hadn't noticed that this review comment wasn't taken into account yet. Anyway, there is no need to take into account changes that will be made by another PR. So feel free to leave this particular change for a future PR.